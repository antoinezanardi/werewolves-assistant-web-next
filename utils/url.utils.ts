function removeTrailingSlashes(url: string): string {
  return url.replace(/\/+$/u, "");
}

export { removeTrailingSlashes };