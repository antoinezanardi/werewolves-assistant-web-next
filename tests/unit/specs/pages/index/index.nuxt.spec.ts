import type { mount } from "@vue/test-utils";
import type { ButtonPassThroughOptions } from "primevue/button";
import type { PassThrough } from "primevue/ts-helpers";

import type { NuxtLink, VuePrimeButton } from "#components";
import Index from "@/pages/index.vue";
import { mountSuspendedComponent } from "~/tests/unit/utils/mount.utils";

describe("Index Page Component", () => {
  let wrapper: ReturnType<typeof mount<typeof Index>>;

  beforeEach(async() => {
    wrapper = await mountSuspendedComponent(Index);
  });

  it("should render component and match snapshot when mounted.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Title", () => {
    it("should display translated title when rendered.", () => {
      const title = wrapper.find("[aria-label='Werewolves Assistant']");

      expect(title.text()).toBe("Werewolves Assistant");
    });
  });

  describe("Play button", () => {
    it("should display play button with translated label when rendered.", () => {
      const playButton = wrapper.findComponent<typeof NuxtLink>("[aria-label='Play']");

      expect(playButton.html()).toContain("Play");
    });
  });

  describe("About button", () => {
    it("should display about button with translated label when rendered.", () => {
      const aboutButton = wrapper.findComponent<typeof NuxtLink>("[aria-label='What is it ?']");

      expect(aboutButton.html()).toContain("What is it ");
    });
  });
});