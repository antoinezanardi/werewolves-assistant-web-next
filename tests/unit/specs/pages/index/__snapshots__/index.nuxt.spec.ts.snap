// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index Page Component > should render component and match snapshot when mounted. 1`] = `
"<div class="container-fluid d-flex flex-column">
  <div class="align-items-center container-fluid d-flex flex-column flex-grow-1 justify-content-center">
    <div class="col-12 text-center">
      <nuxt-img-stub src="/img/logo/square/werewolves-logo.png" sizes="lg:200px" alt="Werewolves Assistant logo"></nuxt-img-stub>
    </div>
    <div class="col-12 text-center">
      <h1 id="title" aria-label="Werewolves Assistant" class="mt-0">Werewolves Assistant</h1>
    </div>
    <div class="col-12 col-lg-3">
      <nuxt-link-stub to="/game-lobby" aria-label="Play" class="p-button p-button-primary p-button-raised w-100"></nuxt-link-stub>
    </div>
    <div class="col-12 col-lg-3 mt-4">
      <nuxt-link-stub to="/about" aria-label="What is it ?" class="p-button p-button-secondary w-100"></nuxt-link-stub>
    </div>
  </div>
  <hr class="w-100">
  <index-footer-stub></index-footer-stub>
</div>"
`;
